src/app/services/main-form-manager/config-map-export-helper.ts

```diff
@@ -484,6 +488,92 @@ export class ConfigMapExportHelper {
     }
   }
 
+  /**
+   * build the correct values from interpolated values.
+   * @param interpolatedValues
+   * @param mode
+   * @param valuesToInsert
+   * @private
+   */
+  private static buildPropsValuesFromInterpolatedValues(interpolatedValues: InterpolatedProperty,
+    mode: LAYER_MODE,
+    valuesToInsert?: (string | number)[]){
+    let interpolatedColor: Array<string | Array<string | number>>;
+    const getField = () =>
+      (interpolatedValues.propertyInterpolatedCountOrMetricCtrl === 'metric')
+        ? interpolatedValues.propertyInterpolatedFieldCtrl + '_' +
+            (interpolatedValues.propertyInterpolatedMetricCtrl as string).toLowerCase() + '_' :
+        interpolatedValues.propertyInterpolatedFieldCtrl;
+
+    if (mode !== LAYER_MODE.features && interpolatedValues.propertyInterpolatedCountOrMetricCtrl === 'count') {
+      // for types FEATURE-METRIC and CLUSTER, if we interpolate by count
+      interpolatedColor = [
+        'interpolate',
+        ['linear'],
+        ['get', 'count' + (!!interpolatedValues.propertyInterpolatedCountNormalizeCtrl ? `_:${NORMALIZED}` : '')]
+      ];
+    } else if (interpolatedValues.propertyInterpolatedNormalizeCtrl) {
+      // otherwise if we normalize
+      interpolatedColor = [
+        'interpolate',
+        ['linear'],
+        this.getArray(
+          getField()
+            .concat(':' + NORMALIZED)
+            .concat(interpolatedValues.propertyInterpolatedNormalizeByKeyCtrl ?
+              ':' + interpolatedValues.propertyInterpolatedNormalizeLocalFieldCtrl : ''))
+      ];
+    } else {
+      // if we don't normalize
+      interpolatedColor = [
+        'interpolate',
+        ['linear'],
+        this.getArray(getField())
+      ];
+    }
+
+    if(valuesToInsert){
+      return interpolatedColor.concat(valuesToInsert);
+    }
+
+    return interpolatedColor;
+  }
+
+  /**
+   *  Methode used to construct the key props
+   */
+  public static getCircleHeatMapSortKey(fgValues: any, mode: LAYER_MODE): PaintValue {
+    switch (fgValues.propertySource) {
+      case PROPERTY_SELECTOR_SOURCE.fix_color:
+        return 1;
+      case PROPERTY_SELECTOR_SOURCE.interpolated:
+        const interpolatedValues = fgValues.propertyInterpolatedFg as InterpolatedProperty;
+        const minValue = interpolatedValues.propertyInterpolatedValuesCtrl[0].proportion;
+        const maxValue = interpolatedValues
+          .propertyInterpolatedValuesCtrl[interpolatedValues.propertyInterpolatedValuesCtrl.length - 1]
+          .proportion;
+        return <PaintValue> this.buildPropsValuesFromInterpolatedValues(interpolatedValues, mode, [minValue, 0, maxValue, 8]);
```

Comment:
Why is the array [min, 0, max, 8] ?

Reply:
its the value i want to add to interpolate my circle sort key

Reply:
Do the values 0 and 8 have a particular meaning, or is it just something that works well ?

Reply:
No particular meaning. Its juste value that works well . Features with a higher sort key will appear above features with a lower sort key

Reply:
@MAudelGisaia could you please add a comment that reminds that those values don't have a special meaning and made to guarantee the interpolation of the circle sort key ?

Reply:
yep i will